src/Reader.ts

```diff
@@ -0,0 +1,485 @@
+      return await this.bookmarkModule.deleteBookmark(bookmark);
+    }
+  };
+  deleteAnnotation = async (highlight) => {
+    return await this.annotationModule?.deleteAnnotation(highlight);
+  };
+  addAnnotation = async (highlight) => {
+    return await this.annotationModule?.addAnnotation(highlight);
+  };
+  tableOfContents = async () => {
+    return await this.navigator.tableOfContents();
+  };
+  readingOrder = async () => {
+    return await this.navigator.readingOrder();
+  };
+  bookmarks = async () => {
+    if (this.navigator.rights?.enableBookmarks) {
+      return await this.bookmarkModule.getBookmarks();
+    } else {
+      return [];
+    }
+  };
+  annotations = async () => {
+    return (await this.annotationModule?.getAnnotations()) ?? [];
+  };
+
+  /**
+   * Search
+   */
+  search = async (term, current) => {
+    if (this.navigator.rights?.enableSearch) {
+      return await this.searchModule?.search(term, current);
+    } else {
+      return [];
+    }
+  };
+  goToSearchIndex = async (href, index, current) => {
+    if (this.navigator.rights?.enableSearch) {
+      await this.searchModule?.goToSearchIndex(href, index, current);
+    }
+  };
+  goToSearchID = async (href, index, current) => {
+    if (this.navigator.rights?.enableSearch) {
+      await this.searchModule?.goToSearchID(href, index, current);
+    }
+  };
+  clearSearch = async () => {
+    if (this.navigator.rights?.enableSearch) {
+      await this.searchModule?.clearSearch();
+    }
+  };
+
+  /**
+   * Resources
+   */
+  currentResource = () => {
+    return this.navigator.currentResource();
+  };
+  mostRecentNavigatedTocItem = () => {
+    return this.navigator.mostRecentNavigatedTocItem();
+  };
+  totalResources = () => {
+    return this.navigator.totalResources();
+  };
+
+  /**
+   * Settings
+   */
+  get publicationLanguage() {
+    return this.navigator.publication.metadata.language;
+  }
+  resetUserSettings = async () => {
+    return await this.settings.resetUserSettings();
+  };
+  applyUserSettings = async (userSettings) => {
+    return await this.settings.applyUserSettings(userSettings);
+  };
+  currentSettings = () => {
+    return this.settings.currentSettings();
+  };
+  scroll = async (value) => {
+    return await this.settings.scroll(value);
+  };
+
+  /**
+   * pitch?
+   */
+  increase = (incremental) => {
+    if (
+      (incremental === "pitch" ||
+        incremental === "rate" ||
+        incremental === "volume") &&
+      this.navigator.rights?.enableTTS
+    ) {
+      this.ttsSettings.increase(incremental);
+    } else {
+      this.settings.increase(incremental);
+    }
+  };
+  decrease = (incremental) => {
+    if (
+      (incremental === "pitch" ||
+        incremental === "rate" ||
+        incremental === "volume") &&
+      this.navigator.rights?.enableTTS
+    ) {
+      this.ttsSettings.decrease(incremental);
+    } else {
+      this.settings.decrease(incremental);
+    }
+  };
+
+  /**
+   * Publisher?
```

Comment:
The opposite of advanced settings on... if the advanced settings are off, we use the default publisher settings.

Reply:
boolean